ChopChop/Storage/StorageManager.swift

```diff
@@ -228,6 +233,172 @@ extension StorageManager {
+
+    func removeFollowee(userId: String, followeeId: String) {
+        firebase.removeFollowee(userId: userId, followeeId: followeeId)
+    }
+
+    func rateRecipe(recipeId: String, userId: String, rating: RatingScore) throws {
+        firebase.addUserRecipeRating(userId: userId, rating: UserRating(recipeOnlineId: recipeId, score: rating))
+        firebase.addRecipeRating(onlineRecipeId: recipeId, rating: RecipeRating(userId: userId, score: rating))
+//        let ratingRecord = RatingRecord(userId: recipeId, recipeId: userId, rating: rating)
+//        try firebase.addRating(rating: ratingRecord)
+    }
+
+//    func unrateRecipe(ratingId: String) {
+//        firebase.removeRating(ratingId: ratingId)
+//    }
+    func unrateRecipe(recipeId: String, rating: RecipeRating) {
+        firebase.removeRecipeRating(onlineRecipeId: recipeId, rating: rating)
+        firebase.removeUserRecipeRating(userId: rating.userId, rating: UserRating(recipeOnlineId: recipeId, score: rating.score))
+    }
+
+//    func rerateRecipe(ratingId: String, newRating: RatingScore) {
+//        firebase.updateRating(ratingId: ratingId, rating: newRating)
+//    }
+    func rerateRecipe(recipeId: String, newRating: RecipeRating) {
+        firebase.updateRecipeRating(userId: newRating.userId, recipeId: recipeId, newScore: newRating.score)
+        firebase.updateUserRating(userId: newRating.userId, recipeId: recipeId, newScore: newRating.score)
+    }
+
+    func fetchAllFriends(userId: String) -> AnyPublisher<[User], Error> {
+        firebase.fetchFriendsId(userId: userId)
+            .flatMap({ followees in
+                firebase.fetchUsers(userId: followees)
+            })
+            .eraseToAnyPublisher()
+    }
+
+    func fetchAllSelfPublishedRecipes(userId: String) -> AnyPublisher<[OnlineRecipe], Error> {
+        firebase.fetchOnlineRecipeIdByUsers(userIds: [userId])
+            .map({
+                $0.compactMap({
+                    try? $0.toOnlineRecipe()
+                })
+            })
+            .eraseToAnyPublisher()
+    }
+
+    func fetchAllRecipes() -> AnyPublisher<[OnlineRecipe], Error> {
+        firebase.fetchAllRecipes()
+            .map({
+                $0.compactMap({
+                    try? $0.toOnlineRecipe()
+                })
+            })
+            .eraseToAnyPublisher()
+    }
+
+    func fetchAllUsers() -> AnyPublisher<[User], Error> {
+        firebase.fetchAllUsers()
+    }
+
+    func fetchAllFolloweeRecipes(followees: [String]) -> AnyPublisher<[OnlineRecipe], Error> {
+        firebase.fetchOnlineRecipeIdByUsers(userIds: followees)
+            .map({
+                $0.compactMap({
+                    try? $0.toOnlineRecipe()
+                })
+            })
+            .eraseToAnyPublisher()
+    }
+    func fetchAllUserRatings(userId: String) -> AnyPublisher<[UserRating], Error> {
+        firebase.fetchUserRating(userId: userId)
+    }
+    func downloadRecipe(recipe: OnlineRecipe) throws {
```

Comment:
can take in a newName: String as well?
cos user needs to input a new name

Reply:
sure so that will be the name for the downloaded recipe right